contactsummary/js/src/contact-summary/controllers/key-dates.controller.js

```diff
@@ -46,47 +48,92 @@ define([
     }
 
     /**
-     * Fetch Contacts and Job Roles from Server
+     * Initialises subscribers
      */
-    function getContacts () {
-      resetKeyDates();
+    function initSubscribers () {
+      $log.debug('Subcsribers initialiazed!');
 
-      Contract.get()
-        .then(function (response) {
-          if (!_.isEmpty(response)) {
-            angular.forEach(response, function (contract) {
-              addContractDates(contract);
+      var events = [
+        'Contract::created',
+        'Contract::updated',
+        'JobRole::created',
+        'JobRole::updated',
+        'JobRole::deleted'
+      ];
 
-              if (contract.is_current === '1') {
-                vm.activeContracts++;
-              }
-            });
-          }
+      events.forEach(function (event) {
+        pubSub.subscribe(event, loadData);
+      });
 
-          return JobRole.get();
-        })
+      pubSub.subscribe('Contract::deleted', function (contract) {
+        Contract.removeContract(contract);
+        loadData();
+      });
```

Comment:
This block is a bit confusing. We listen for contract deletion and then remove the contract?

Also, why is a simple "key-dates.controller.js" file deleting contracts?

Reply:
Hm... this is an existing code (https://github.com/compucorp/civihr/pull/2958/files#diff-4c0fd0b30d001fefdb6b6611f6f8b41dL119). But let me check, good shout.